fluent/fluent.go

```diff
@@ -952,6 +984,89 @@ func (n *nextHopEntry) WithEncapsulateHeader(h Header) *nextHopEntry {
 	return n
 }
 
+func (n *nextHopEntry) nextEncapHeaderKeyIndex() uint64 {
```

Comment:
(nit) Can you add a doc string here.

```suggestion
// nextEncapHeaderKeyIndex returns the index of the next encapsulation to be added to
// the encap-headers list for the specified next-hop entry.
func (n *nextHopEntry) nextEncapHeaderKeyIndex() uint64 {
```

Reply:
Done